src/sources/http_server.rs

```diff
@@ -568,15 +568,19 @@ impl HttpSource for SimpleHttpSource {
     /// This method adds the custom headers specified in the configuration
     /// to the HTTP response.
     fn enrich_reply<T: warp::Reply + 'static>(&self, reply: T) -> Box<dyn warp::Reply> {
-        let mut boxed_reply: Box<dyn warp::Reply> = Box::new(reply);
+        let mut response = reply.into_response();
+        let header_map = response.headers_mut();
+
         for (key, values) in &self.custom_response_headers {
-            boxed_reply = Box::new(warp::reply::with_header(
-                boxed_reply,
-                key,
-                values.join(", "),
-            ));
+            let header_name: HeaderName = key.parse().unwrap();
+            if let Some((first, rest)) = values.split_first() {
+                header_map.insert(header_name.clone(), first.parse().unwrap());
```

Comment:
I think you can just use `append` rather than needing to split and `insert` the first one:

> Inserts a key-value pair into the map.
> 
> If the map did not previously have this key present, then false is returned.
> 
> If the map did have this key present, the new value is pushed to the end of the list of values currently associated with the key. The key is not updated, though; this matters for types that can be == without being identical.

https://docs.rs/http/0.2.12/http/header/struct.HeaderMap.html#method.append

Reply:
From the docs you point, I understand that if you append without inserting first, it would fail

Thats why I suggested to insert the first and append the rest

Reply:
I don't think it fails. It just returns `false` if the key didn't already exist and `true` if it did but either way the value is inserted.

See the implementation: https://docs.rs/http/0.2.12/src/http/header/map.rs.html#1415-1449

Reply:
Oh, you're right, didn't see the implementation. I think that is a bit misleading from the documentation, at least, how I understood it..

Reply:
For sure :) I can see why you would have been confused.